slides/advanced_topics_slides.md

```diff
@@ -0,0 +1,159 @@
+class: center, middle
+
+# Advanced topics: What to try next?
+
+So far this course covered the basics of machine learning. 
+
+This presentation gives an overview of what is out there for you to explore
+in the field of machine learning. 
+
+???
+
+We throw a lot of different topics at you, it is up to you to decide 
+which you want to pick up.
+The sky is the limit! Possible side-effects: You may feel slightly overwhelmed. 
+Just take it one step at a time.
+
+---
+
+# Machine learning algorithms
+There are many different machine learning algorithms that you can explore.
+
+???
+A short recap and further details about different machine learning models
+
+---
+# Linear models
+<img src="../figures/linear_fit_red.svg" width="60%">
+
+???
+
+Linear models: fit a straight line through the data that minimizes the prediction error.
+
+---
+
+# Decision trees
+<img src="../figures/tree_example.svg" width="100%" style="float: right">
+
+???
+
+Decision trees: Decision trees make decisions by asking a series of questions and choosing the best path based on the answers.
+The 'decisions' are based on the patterns in the data.
+
+---
+
+# Support Vector Machines (SVMs)
+<img src="../figures/svm.png" width="50%" >
+
+???
+
+SVM finds the best way to separate different classes in data by creating a clear gap between them.
+
+---
+
+# Ensemble learning
+<img src="../figures/ensemble-learning.png" width="50%" style="float: left">
+
+???
+
+Combine different (weak) classifiers by aggregating their individual outputs into one prediction.
+Examples are: random forests or gradient boosted decision trees
+
+---
+
+# Neural networks
+<img src="../figures/neural-network.png" width="100%" style="float: right">
+
+???
+
+Neural networks consist of artificial neurons: small units that integrate information in a simple, yet adaptable way. 
+Through combining a lot of neurons in different layers the neural network can learn patterns in the data.
+
+---
+
+# Deep learning
+<img src="../figures/deep-learning.png" width="100%" style="float: right">
+
+From https://www.kaggle.com/keras/resnet50
+
+???
+
+In deep learning we use complex neural networks with multiple layers. 
+The architecture of the neural network is adapted to specific problems.
+
+---
+
+# Deep learning with fast.ai
+<img src="../figures/fast-ai.png" width="100%" style="float: right">
+
+???
+
+Fast.ai is a library that is designed to 'make neural networks uncool again'.
+It allows you to apply deep learning with a relatively simple syntax,
+that is similar to scikit-learn. 
+Here you see an example of how to train a relatively complex neural network on the adult census dataset.
+
+---
+
+# Model sweep: pick the best model from a set of models
+<img src="../figures/model-sweep.png" width="100%" style="float: right">
+
+???
+
+Typically, you do not know which model 'family' works best on your data.
+It can help to do a quick 'model sweep': try out a few models to see which works best.
```

Comment:
I have a bit of headache with this advice. I see the idea, but applying various models with arbitrary parameters can also yield misleading results, especially in case novices that do not understand the underlying algorithms.

Reply:
Hmmm, I agree there is some nuance to it. But trying out a few models with default parameters (which are default for a reason) will give you an idea of what could work well quickly. I used it a lot in practice. 

The alternative for novices would be to try out different models manually or do a gigantic model sweep and hyperparameter tuning experiment.

But I will add a note for the instructor:

Note that applying various models with arbitrary parameters can also yield misleading results, especially in case novices that do not understand the underlying algorithms. But trying out a few models with default parameters (which are default for a reason) can quickly give you an idea of what could work well.

Reply:
Can you live with that @carschno ? 🙀

Reply:
Yes, but I'm afraid you have taken over a typo that I introduced in my comment. Fixed and stream-lined version:

> Note that applying models with arbitrary parameters can yield misleading results, especially when you do not understand the underlying algorithm.

Reply:
Woops went to fast there! I fixed it.